lib/shopify_api/graphql/response.ex

```diff
@@ -37,7 +37,8 @@ defmodule ShopifyAPI.GraphQL.Response do
   def handle({:error, _} = response, _), do: response
 
   @doc false
-  def build_response(%{body: %{"data" => data, "extensions" => extensions}} = response) do
+  def build_response(%{body: %{"data" => data, "extensions" => extensions}} = response)
+      when data !== nil do
```

Comment:
🍎 You could build this as a match entirely

`def build_response(%{body: %{"data" => nil, "extensions" => extensions}} = response)`

Reply:
If using the guard it should probably be `!=` instead of `!==`